spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -352,12 +353,12 @@ public TopicAdminClient topicAdminClient(
 	@Bean
 	@ConditionalOnMissingBean
 	public TopicAdminSettings topicAdminSettings(
-			TransportChannelProvider transportChannelProvider) {
+			@Qualifier("publisherTransportChannelProvider") TransportChannelProvider publisherTransportChannelProvider) {
```

Comment:
I wonder if admin functionality should have its own. The most obvious usecase is needing different credentials for admin operations.

Reply:
I thought that too, but I saw that they use exactly the same default for admin client.
https://github.com/googleapis/java-pubsub/blob/8cf77ee3b31356f342942a9ffaa61db2d0686769/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriptionAdminSettings.java#L227